gotham/src/clone_sockets.rs

```diff
@@ -30,30 +31,35 @@ where
     for _ in 0..threads - 1 {
         let listener = listener.try_clone().expect("unable to clone TCP listener");
         let protocol = protocol.clone();
-        let service = service.clone();
-        thread::spawn(move || serve(listener, &addr, &protocol, &service));
+        let new_handler = new_handler.clone();
+        thread::spawn(move || serve(listener, &addr, &protocol, new_handler));
     }
 
-    serve(listener, &addr, &protocol, &service);
+    serve(listener, &addr, &protocol, new_handler);
 }
 
-fn serve<NH>(
-    listener: TcpListener,
-    addr: &SocketAddr,
-    protocol: &Http,
-    new_service: &NewHandlerService<NH>,
-) where
+fn serve<NH>(listener: TcpListener, addr: &SocketAddr, protocol: &Http, new_handler: Arc<NH>)
+where
     NH: NewHandler + 'static,
 {
     let mut core = Core::new().expect("unable to spawn tokio reactor");
     let handle = core.handle();
 
+    let new_service = GothamService::new(new_handler, handle.clone());
```

Comment:
Nit: I feel it may help readability to `s/new_service/gotham_service`.

Thoughts?

Reply:
Likewise https://github.com/gotham-rs/gotham/blob/481a912490881f3c9016db6a48d2a53c00406c41/gotham/src/socket_queue.rs#L109